### ISSUE TITLE
SCRIPT2343: Stack overflow on IE11 / Microsoft Edge

### ISSUE THREAD

**Issue by link89** [DevExpress/testcafe-react-selectors#119]
Following Error will be throw when running cases on IE11 / Edge
```
   1) An error occurred in Selector code:

      Error: Out of stack space

      Browser: IE 11.0.0 / Windows 10.0.0
```
If not using react-selector, then everything is ok.

Here is a minimal reproduce for this bug:
```javascript
import 'testcafe';
import { waitForReact, ReactSelector } from 'testcafe-react-selectors';

fixture('semantic')
    .page(`https://react.semantic-ui.com/elements/button/`)

test(`using react selector`, async (t) => {
    await waitForReact();
    const el = ReactSelector('div');
    await t
        .expect(el.count).eql(1);
})
```
And result with detail log is here: 
[react-ie-edge-bug.txt](https://github.com/DevExpress/testcafe-react-selectors/files/2378270/react-ie-edge-bug.txt)

It seems like some kind of polyfill things common on IE11/Edge.

**Comment by kirovboris** [comment]
Hi @link89,

Unfortunately, I can't reproduce the error `"Error: Out of stack space"`. Could you please specify the versions of `testcafe` and `testcafe-react-selectors`? 
I get the following error message in IE 11, in MS Edge the test passes.
```
 'assign' is not a function
```
Also the provided page throws an exception in IE11 without testcafe.

![image](https://user-images.githubusercontent.com/3633477/45482203-0f814280-b756-11e8-8d7d-76e708a2a21d.png)

**Comment by link89** [comment]
@kirovboris Thank you for your quick response. I guess the reason may be the ie version we used is too old. We use an win10 vagrant box download from https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/ , whose release date is 2015.

**Comment by link89** [comment]
@kirovboris  The problem still exists after upgrade both ie11 and edge to latest version. Here is another script to reproduce this bug.
```javascript
import 'testcafe';
import { waitForReact, ReactSelector } from 'testcafe-react-selectors';


fixture('uibench')
    .page(`https://localvoid.github.io/uibench/`)

test(`using react selector`, async (t) => {
    await waitForReact();
    const el = ReactSelector('t div div div div div input').withProps('id', 'test-filter');
    await t.typeText(el, 'hello world');
    await t.wait(5e3);
})
```
It works fine on both chrome and firefox, but fail to run on ie11 and edge.

**Comment by link89** [comment]
@kirovboris Here is the detail log.
[edge-out-of-stack.log](https://github.com/DevExpress/testcafe-react-selectors/files/2387481/edge-out-of-stack.log)

**Comment by kirovboris** [comment]
I still can't reproduce this issue.
Could you please provide some details:
- the testcafe version
- testcafe run options, e.g. I ran testcafe without any options: `testcafe ie test.js`
- the nodejs version

Also if this issue is reproduced only on the virtual machine, could you try to do the same locally or on another computer?

**Comment by link89** [comment]
I see, it's the problem of testcafe 0.22.1-alpha https://github.com/DevExpress/testcafe/releases/tag/v0.22.1-alpha.1

**Comment by AndreyBelym** [comment]
@link89, I've investigated the problem, and it's definitely caused by a bug in the `testcafe-hammerhead` package: https://github.com/DevExpress/testcafe-hammerhead/issues/1763. I will inform you as soon as a fix is available and the fixed TestCafe version is published to NPM.

**Comment by link89** [comment]
@AndreyBelym @kirovboris Thanks. You guys are doing a great job!

**Comment by link89** [comment]
@AndreyBelym Are those fixes included in v0.23.0 ?

**Comment by AndreyBelym** [comment]
@link89, yes, it's included in the `testcafe@0.23.0`, you can give it a try!
